waku/waku_rln_relay/conversion_utils.nim

```diff
@@ -53,33 +53,60 @@ proc encodeLengthPrefix*(input: openArray[byte]): seq[byte] =
     output = concat(@len, @input)
   return output
 
-proc serialize*(idSecretHash: IdentitySecretHash, 
+when defined(rln_v2):
```

Comment:
had to dupe the function because args are different

Reply:
have the hunch that having rlnv2 behind a flag will end up being more problematic than using a branch (where we completely replace v1 with v2)